In [8]:
import numpy as np
import pandas as pd

In [9]:
# read the data :
data = pd.read_csv("FL_insurance_sample.csv")
data

policyID statecode           county  eq_site_limit  hu_site_limit  \
0        119736        FL      CLAY COUNTY       498960.0      498960.00   
1        448094        FL      CLAY COUNTY      1322376.3     1322376.30   
2        206893        FL      CLAY COUNTY       190724.4      190724.40   
3        333743        FL      CLAY COUNTY            0.0       79520.76   
4        172534        FL      CLAY COUNTY            0.0      254281.50   
...         ...       ...              ...            ...            ...   
36629    404309        FL  PINELLAS COUNTY            0.0       88722.00   
36630    465875        FL  PINELLAS COUNTY      1297057.5     1297057.50   
36631    791209        FL  PINELLAS COUNTY       173286.9      173286.90   
36632    322627        FL  PINELLAS COUNTY      1499781.6     1499781.60   
36633    398149        FL  PINELLAS COUNTY       373488.3      373488.30   

       fl_site_limit  fr_site_limit    tiv_2011    tiv_2012  \
0           498960.0       498960.0   498960.00   792148.90   
1          1322376.3      1322376.3  1322376.30  1438163.57   
2           190724.4       190724.4   190724.40   192476.78   
3                0.0            0.0    79520.76    86854.48   
4                0.0       254281.5   254281.50   246144.49   
...              ...            ...         ...         ...   
36629            0.0        88722.0    88722.00   155065.65   
36630      1297057.5      1297057.5  1297057.50  1577500.79   
36631            0.0            0.0   173286.90   205527.97   
36632            0.0            0.0  1499781.60  1768782.43   
36633            0.0            0.0   373488.30   596003.67   

       eq_site_deductible  hu_site_deductible  fl_site_deductible  \
0                     0.0              9979.2                 0.0   
1                     0.0                 0.0                 0.0   
2                     0.0                 0.0                 0.0   
3                     0.0                 0.0                 0.0   
4                     0.0                 0.0                 0.0   
...                   ...                 ...                 ...   
36629                 0.0                 0.0                 0.0   
36630                 0.0                 0.0                 0.0   
36631                 0.0                 0.0                 0.0   
36632                 0.0                 0.0                 0.0   
36633                 0.0                 0.0                 0.0   

       fr_site_deductible  point_latitude  point_longitude         line  \
0                       0       30.102261       -81.711777  Residential   
1                       0       30.063936       -81.707664  Residential   
2                       0       30.089579       -81.700455  Residential   
3                       0       30.063236       -81.707703  Residential   
4                       0       30.060614       -81.702675  Residential   
...                   ...             ...              ...          ...   
36629                   0       28.122885       -82.770218  Residential   
36630                   0       28.080900       -82.758800  Residential   
36631                   0       28.110550       -82.766360  Residential   
36632                   0       28.089415       -82.697411  Residential   
36633                   0       28.064440       -82.774590  Residential   

      construction  point_granularity  
0          Masonry                  1  
1          Masonry                  3  
2             Wood                  1  
3             Wood                  3  
4             Wood                  1  
...            ...                ...  
36629         Wood                  3  
36630      Masonry                  1  
36631         Wood                  4  
36632      Masonry                  3  
36633      Masonry                  1  

[36634 rows x 18 columns]

In [10]:
# checking type of data : 

type(data)

pandas.core.frame.DataFrame

In [11]:
# to check the shape of the data :
print(data.shape)
# to get the number of rows or instances or records or examples
print(data.shape[0])
# to get the number of columns or features or attributes
print(data.shape[1])

(36634, 18)
36634
18


In [14]:
# to get the column names:
print(data.columns)
print()
# to print the rows names:
print(data.index)

Index(['policyID', 'statecode', 'county', 'eq_site_limit', 'hu_site_limit',
       'fl_site_limit', 'fr_site_limit', 'tiv_2011', 'tiv_2012',
       'eq_site_deductible', 'hu_site_deductible', 'fl_site_deductible',
       'fr_site_deductible', 'point_latitude', 'point_longitude', 'line',
       'construction', 'point_granularity'],
      dtype='object')

RangeIndex(start=0, stop=36634, step=1)


In [21]:
# datatype of each column
print(data.dtypes)
print("___"*15)
# what is output type for data.dtypes:
print(type(data.dtypes))
print("___"*15)
#acsessing the first column datatype:
print(data.dtypes[0])
print("___"*15)
# acess the last column datatype :
print(data.dtypes[-1])

policyID                int64
statecode              object
county                 object
eq_site_limit         float64
hu_site_limit         float64
fl_site_limit         float64
fr_site_limit         float64
tiv_2011              float64
tiv_2012              float64
eq_site_deductible    float64
hu_site_deductible    float64
fl_site_deductible    float64
fr_site_deductible      int64
point_latitude        float64
point_longitude       float64
line                   object
construction           object
point_granularity       int64
dtype: object
_____________________________________________
<class 'pandas.core.series.Series'>
_____________________________________________
int64
_____________________________________________
int64


In [30]:
# column count for each datatype:
# it will apply on the series not on dataframe

data.dtypes.value_counts()

float64    11
object      4
int64       3
dtype: int64

In [23]:
# to display top n rows:
data.head(10)

policyID statecode       county  eq_site_limit  hu_site_limit  \
0    119736        FL  CLAY COUNTY       498960.0      498960.00   
1    448094        FL  CLAY COUNTY      1322376.3     1322376.30   
2    206893        FL  CLAY COUNTY       190724.4      190724.40   
3    333743        FL  CLAY COUNTY            0.0       79520.76   
4    172534        FL  CLAY COUNTY            0.0      254281.50   
5    785275        FL  CLAY COUNTY            0.0      515035.62   
6    995932        FL  CLAY COUNTY            0.0    19260000.00   
7    223488        FL  CLAY COUNTY       328500.0      328500.00   
8    433512        FL  CLAY COUNTY       315000.0      315000.00   
9    142071        FL  CLAY COUNTY       705600.0      705600.00   

   fl_site_limit  fr_site_limit     tiv_2011     tiv_2012  eq_site_deductible  \
0       498960.0       498960.0    498960.00    792148.90                 0.0   
1      1322376.3      1322376.3   1322376.30   1438163.57                 0.0   
2       190724.4       190724.4    190724.40    192476.78                 0.0   
3            0.0            0.0     79520.76     86854.48                 0.0   
4            0.0       254281.5    254281.50    246144.49                 0.0   
5            0.0            0.0    515035.62    884419.17                 0.0   
6            0.0            0.0  19260000.00  20610000.00                 0.0   
7       328500.0       328500.0    328500.00    348374.25                 0.0   
8       315000.0       315000.0    315000.00    265821.57                 0.0   
9       705600.0       705600.0    705600.00   1010842.56             14112.0   

   hu_site_deductible  fl_site_deductible  fr_site_deductible  point_latitude  \
0              9979.2                 0.0                   0       30.102261   
1                 0.0                 0.0                   0       30.063936   
2                 0.0                 0.0                   0       30.089579   
3                 0.0                 0.0                   0       30.063236   
4                 0.0                 0.0                   0       30.060614   
5                 0.0                 0.0                   0       30.063236   
6                 0.0                 0.0                   0       30.102226   
7             16425.0                 0.0                   0       30.102217   
8             15750.0                 0.0                   0       30.118774   
9             35280.0                 0.0                   0       30.100628   

   point_longitude         line         construction  point_granularity  
0       -81.711777  Residential              Masonry                  1  
1       -81.707664  Residential              Masonry                  3  
2       -81.700455  Residential                 Wood                  1  
3       -81.707703  Residential                 Wood                  3  
4       -81.702675  Residential                 Wood                  1  
5       -81.707703  Residential              Masonry                  3  
6       -81.713882   Commercial  Reinforced Concrete                  1  
7       -81.707146  Residential                 Wood                  1  
8       -81.704613  Residential                 Wood                  1  
9       -81.703751  Residential              Masonry                  1

In [24]:
# to display last n rows:
data.tail(10)

policyID statecode           county  eq_site_limit  hu_site_limit  \
36624    827844        FL  PINELLAS COUNTY            0.0        38112.3   
36625    876385        FL  PINELLAS COUNTY            0.0        18312.3   
36626    167630        FL  PINELLAS COUNTY            0.0         4522.5   
36627    241496        FL  PINELLAS COUNTY            0.0         1350.0   
36628    780814        FL  PINELLAS COUNTY            0.0         9770.4   
36629    404309        FL  PINELLAS COUNTY            0.0        88722.0   
36630    465875        FL  PINELLAS COUNTY      1297057.5      1297057.5   
36631    791209        FL  PINELLAS COUNTY       173286.9       173286.9   
36632    322627        FL  PINELLAS COUNTY      1499781.6      1499781.6   
36633    398149        FL  PINELLAS COUNTY       373488.3       373488.3   

       fl_site_limit  fr_site_limit   tiv_2011    tiv_2012  \
36624            0.0        38112.3    38112.3    52710.45   
36625            0.0        18312.3    18312.3    18575.81   
36626            0.0         4522.5     4522.5     4048.11   
36627            0.0         1350.0     1350.0     1430.16   
36628            0.0         9770.4     9770.4     9349.98   
36629            0.0        88722.0    88722.0   155065.65   
36630      1297057.5      1297057.5  1297057.5  1577500.79   
36631            0.0            0.0   173286.9   205527.97   
36632            0.0            0.0  1499781.6  1768782.43   
36633            0.0            0.0   373488.3   596003.67   

       eq_site_deductible  hu_site_deductible  fl_site_deductible  \
36624                 0.0                 0.0                 0.0   
36625                 0.0                 0.0                 0.0   
36626                 0.0                 0.0                 0.0   
36627                 0.0                 0.0                 0.0   
36628                 0.0                 0.0                 0.0   
36629                 0.0                 0.0                 0.0   
36630                 0.0                 0.0                 0.0   
36631                 0.0                 0.0                 0.0   
36632                 0.0                 0.0                 0.0   
36633                 0.0                 0.0                 0.0   

       fr_site_deductible  point_latitude  point_longitude         line  \
36624                   0       28.122885       -82.770218  Residential   
36625                   0       28.122885       -82.770218  Residential   
36626                   0       28.122885       -82.770218  Residential   
36627                   0       28.122885       -82.770218  Residential   
36628                   0       28.122885       -82.770218  Residential   
36629                   0       28.122885       -82.770218  Residential   
36630                   0       28.080900       -82.758800  Residential   
36631                   0       28.110550       -82.766360  Residential   
36632                   0       28.089415       -82.697411  Residential   
36633                   0       28.064440       -82.774590  Residential   

      construction  point_granularity  
36624         Wood                  3  
36625         Wood                  3  
36626         Wood                  3  
36627         Wood                  3  
36628         Wood                  3  
36629         Wood                  3  
36630      Masonry                  1  
36631         Wood                  4  
36632      Masonry                  3  
36633      Masonry                  1

In [25]:
# select random n rows:
data.sample(5)

policyID statecode             county  eq_site_limit  hu_site_limit  \
6452     809607        FL     JACKSON COUNTY            0.0      894803.44   
126      363828        FL    COLUMBIA COUNTY            0.0      132603.34   
29848    996240        FL        POLK COUNTY            0.0       10177.61   
36364    814376        FL       PASCO COUNTY            0.0      338690.39   
23534    414146        FL  PALM BEACH COUNTY            0.0       49006.98   

       fl_site_limit  fr_site_limit   tiv_2011    tiv_2012  \
6452             0.0            0.0  894803.44  1251472.09   
126              0.0            0.0  132603.34   216561.94   
29848            0.0            0.0   10177.61    15578.82   
36364            0.0            0.0  338690.39   478612.19   
23534            0.0            0.0   49006.98    40883.58   

       eq_site_deductible  hu_site_deductible  fl_site_deductible  \
6452                  0.0                 0.0                 0.0   
126                   0.0                 0.0                 0.0   
29848                 0.0                 0.0                 0.0   
36364                 0.0                 0.0                 0.0   
23534                 0.0                 0.0                 0.0   

       fr_site_deductible  point_latitude  point_longitude         line  \
6452                    0       30.949102       -85.515793  Residential   
126                     0       30.176380       -82.663140  Residential   
29848                   0       27.965130       -81.885710  Residential   
36364                   0       28.217358       -82.666199  Residential   
23534                   0       26.823654       -80.105865  Residential   

      construction  point_granularity  
6452       Masonry                  1  
126           Wood                  3  
29848         Wood                  1  
36364         Wood                  1  
23534         Wood                  1

In [26]:
# to get the info about the dataframe : 
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36634 entries, 0 to 36633
Data columns (total 18 columns):
policyID              36634 non-null int64
statecode             36634 non-null object
county                36634 non-null object
eq_site_limit         36634 non-null float64
hu_site_limit         36634 non-null float64
fl_site_limit         36634 non-null float64
fr_site_limit         36634 non-null float64
tiv_2011              36634 non-null float64
tiv_2012              36634 non-null float64
eq_site_deductible    36634 non-null float64
hu_site_deductible    36634 non-null float64
fl_site_deductible    36634 non-null float64
fr_site_deductible    36634 non-null int64
point_latitude        36634 non-null float64
point_longitude       36634 non-null float64
line                  36634 non-null object
construction          36634 non-null object
point_granularity     36634 non-null int64
dtypes: float64(11), int64(3), object(4)
memory usage: 5.0+ MB


In [27]:
type(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36634 entries, 0 to 36633
Data columns (total 18 columns):
policyID              36634 non-null int64
statecode             36634 non-null object
county                36634 non-null object
eq_site_limit         36634 non-null float64
hu_site_limit         36634 non-null float64
fl_site_limit         36634 non-null float64
fr_site_limit         36634 non-null float64
tiv_2011              36634 non-null float64
tiv_2012              36634 non-null float64
eq_site_deductible    36634 non-null float64
hu_site_deductible    36634 non-null float64
fl_site_deductible    36634 non-null float64
fr_site_deductible    36634 non-null int64
point_latitude        36634 non-null float64
point_longitude       36634 non-null float64
line                  36634 non-null object
construction          36634 non-null object
point_granularity     36634 non-null int64
dtypes: float64(11), int64(3), object(4)
memory usage: 5.0+ MB


NoneType

In [31]:
# acessing the single column
data.columns

Index(['policyID', 'statecode', 'county', 'eq_site_limit', 'hu_site_limit',
       'fl_site_limit', 'fr_site_limit', 'tiv_2011', 'tiv_2012',
       'eq_site_deductible', 'hu_site_deductible', 'fl_site_deductible',
       'fr_site_deductible', 'point_latitude', 'point_longitude', 'line',
       'construction', 'point_granularity'],
      dtype='object')

In [35]:
data.policyID  # or data['policyID']

0        119736
1        448094
2        206893
3        333743
4        172534
          ...  
36629    404309
36630    465875
36631    791209
36632    322627
36633    398149
Name: policyID, Length: 36634, dtype: int64

In [34]:
type(data.policyID)

pandas.core.series.Series

In [40]:
# specific column datatype:

data.policyID.dtypes

dtype('int64')

#### difference btwn dtype and dtypes:
- dataframe object doesnot have dtype

- numerical columns - int or float data
- categorical columns - object type(string data)
- bollean data - true or false

In [41]:
data.point_granularity

0        1
1        3
2        1
3        3
4        1
        ..
36629    3
36630    1
36631    4
36632    3
36633    1
Name: point_granularity, Length: 36634, dtype: int64

In [42]:
type(data.point_granularity)

pandas.core.series.Series

In [43]:
data.point_granularity.dtype

dtype('int64')

In [45]:
data.point_granularity.mean()

1.6409073538243162

In [46]:
data.point_granularity.sum()/data.point_granularity.shape[0]

1.6409073538243162

In [48]:
# minimum value of a column
# arg min will give the index of the min value
print(data.point_granularity.min())
# maximum value of a column
# argmax will give the index of the max value
print(data.point_granularity.max())

1
7


In [49]:
data.point_granularity.value_counts()

1    26569
3     6724
4     3337
7        2
5        2
Name: point_granularity, dtype: int64

In [50]:
# output type is series :
data.point_granularity.mode()

0    1
dtype: int64

In [51]:
data.point_granularity.mode()[0]

1

In [52]:
data.point_granularity.median()

1.0

In [54]:
# range of values in a column
(data.point_granularity.min(),data.point_granularity.max())

(1, 7)

In [57]:
data.columns

Index(['policyID', 'statecode', 'county', 'eq_site_limit', 'hu_site_limit',
       'fl_site_limit', 'fr_site_limit', 'tiv_2011', 'tiv_2012',
       'eq_site_deductible', 'hu_site_deductible', 'fl_site_deductible',
       'fr_site_deductible', 'point_latitude', 'point_longitude', 'line',
       'construction', 'point_granularity'],
      dtype='object')

In [59]:
data.county.dtype   # it is object type

dtype('O')

In [60]:
data.county.count()

36634

In [71]:
print(data.county.value_counts())
print("_"*14)
print(type(data.county.value_counts()))
# type is series

MIAMI DADE COUNTY    4315
BROWARD COUNTY       3193
PALM BEACH COUNTY    2791
DUVAL COUNTY         1894
ORANGE COUNTY        1811
                     ... 
GLADES COUNTY          22
UNION COUNTY           15
Orlando                 1
OSCEOLA COUNTY          1
North Fort Myers        1
Name: county, Length: 67, dtype: int64
______________
<class 'pandas.core.series.Series'>


In [72]:
# unique values:
print(data.county.unique()) # or data.county.value_counts().index
# unique values count
print(data.county.nunique())

['CLAY COUNTY' 'SUWANNEE COUNTY' 'NASSAU COUNTY' 'COLUMBIA COUNTY'
 'ST  JOHNS COUNTY' 'BAKER COUNTY' 'BRADFORD COUNTY' 'HAMILTON COUNTY'
 'UNION COUNTY' 'MADISON COUNTY' 'LAFAYETTE COUNTY' 'FLAGLER COUNTY'
 'DUVAL COUNTY' 'LAKE COUNTY' 'VOLUSIA COUNTY' 'PUTNAM COUNTY'
 'MARION COUNTY' 'SUMTER COUNTY' 'LEON COUNTY' 'FRANKLIN COUNTY'
 'LIBERTY COUNTY' 'GADSDEN COUNTY' 'WAKULLA COUNTY' 'JEFFERSON COUNTY'
 'TAYLOR COUNTY' 'BAY COUNTY' 'WALTON COUNTY' 'JACKSON COUNTY'
 'CALHOUN COUNTY' 'HOLMES COUNTY' 'WASHINGTON COUNTY' 'GULF COUNTY'
 'ESCAMBIA COUNTY' 'SANTA ROSA COUNTY' 'OKALOOSA COUNTY' 'ALACHUA COUNTY'
 'GILCHRIST COUNTY' 'LEVY COUNTY' 'DIXIE COUNTY' 'SEMINOLE COUNTY'
 'ORANGE COUNTY' 'BREVARD COUNTY' 'INDIAN RIVER COUNTY'
 'MIAMI DADE COUNTY' 'BROWARD COUNTY' 'MONROE COUNTY' 'PALM BEACH COUNTY'
 'MARTIN COUNTY' 'HENDRY COUNTY' 'PASCO COUNTY' 'GLADES COUNTY'
 'HILLSBOROUGH COUNTY' 'HERNANDO COUNTY' 'PINELLAS COUNTY' 'POLK COUNTY'
 'North Fort Myers' 'Orlando' 'HIGHLANDS COUNTY' 'HARDE

In [73]:
type(data.county.nunique())

int

In [74]:
data.describe()

policyID  eq_site_limit  hu_site_limit  fl_site_limit  \
count   36634.000000   3.663400e+04   3.663400e+04   3.663400e+04   
mean   548661.937189   7.314780e+05   2.074348e+06   6.646009e+05   
std    259912.926494   1.752300e+07   1.964150e+07   1.746114e+07   
min    100074.000000   0.000000e+00   0.000000e+00   0.000000e+00   
25%    323647.750000   0.000000e+00   4.248106e+04   0.000000e+00   
50%    548525.000000   0.000000e+00   1.926911e+05   0.000000e+00   
75%    774252.000000   0.000000e+00   9.163381e+05   0.000000e+00   
max    999971.000000   2.160000e+09   2.160000e+09   2.160000e+09   

       fr_site_limit      tiv_2011      tiv_2012  eq_site_deductible  \
count   3.663400e+04  3.663400e+04  3.663400e+04        3.663400e+04   
mean    9.911724e+05  2.172875e+06  2.571004e+06        7.787908e+02   
std     1.767178e+07  2.015015e+07  2.036777e+07        4.787979e+04   
min     0.000000e+00  9.000000e+01  7.337000e+01        0.000000e+00   
25%     0.000000e+00  4.562415e+04  5.401351e+04        0.000000e+00   
50%     0.000000e+00  2.021051e+05  2.416310e+05        0.000000e+00   
75%     5.741010e+04  9.465821e+05  1.128871e+06        0.000000e+00   
max     2.160000e+09  2.160000e+09  1.701000e+09        6.273765e+06   

       hu_site_deductible  fl_site_deductible  fr_site_deductible  \
count        3.663400e+04        36634.000000        36634.000000   
mean         7.037985e+03          192.453210           26.483594   
std          9.582593e+04         8037.713716         4702.740932   
min          0.000000e+00            0.000000            0.000000   
25%          0.000000e+00            0.000000            0.000000   
50%          0.000000e+00            0.000000            0.000000   
75%          0.000000e+00            0.000000            0.000000   
max          7.380000e+06       450000.000000       900000.000000   

       point_latitude  point_longitude  point_granularity  
count    36634.000000     36634.000000       36634.000000  
mean        28.087477       -81.903565           1.640907  
std          1.647734         1.809066           1.070561  
min         24.547514       -87.447290           1.000000  
25%         26.471996       -82.439331           1.000000  
50%         28.057077       -81.585700           1.000000  
75%         29.623005       -80.372910           3.000000  
max         30.989820       -80.033257           7.000000

In [75]:
data.describe().columns

Index(['policyID', 'eq_site_limit', 'hu_site_limit', 'fl_site_limit',
       'fr_site_limit', 'tiv_2011', 'tiv_2012', 'eq_site_deductible',
       'hu_site_deductible', 'fl_site_deductible', 'fr_site_deductible',
       'point_latitude', 'point_longitude', 'point_granularity'],
      dtype='object')

In [81]:
#  to get the description for the categorical columns
data.describe(include=object)

statecode             county         line construction
count      36634              36634        36634        36634
unique         1                 67            2            5
top           FL  MIAMI DADE COUNTY  Residential         Wood
freq       36634               4315        30838        21581

In [77]:
# to get the description for the categorical columns by using int
data.describe(include=np.int64)

policyID  fr_site_deductible  point_granularity
count   36634.000000        36634.000000       36634.000000
mean   548661.937189           26.483594           1.640907
std    259912.926494         4702.740932           1.070561
min    100074.000000            0.000000           1.000000
25%    323647.750000            0.000000           1.000000
50%    548525.000000            0.000000           1.000000
75%    774252.000000            0.000000           3.000000
max    999971.000000       900000.000000           7.000000

In [78]:
# to get the description for the categorical columns by using float
data.describe(include=np.float64)

eq_site_limit  hu_site_limit  fl_site_limit  fr_site_limit  \
count   3.663400e+04   3.663400e+04   3.663400e+04   3.663400e+04   
mean    7.314780e+05   2.074348e+06   6.646009e+05   9.911724e+05   
std     1.752300e+07   1.964150e+07   1.746114e+07   1.767178e+07   
min     0.000000e+00   0.000000e+00   0.000000e+00   0.000000e+00   
25%     0.000000e+00   4.248106e+04   0.000000e+00   0.000000e+00   
50%     0.000000e+00   1.926911e+05   0.000000e+00   0.000000e+00   
75%     0.000000e+00   9.163381e+05   0.000000e+00   5.741010e+04   
max     2.160000e+09   2.160000e+09   2.160000e+09   2.160000e+09   

           tiv_2011      tiv_2012  eq_site_deductible  hu_site_deductible  \
count  3.663400e+04  3.663400e+04        3.663400e+04        3.663400e+04   
mean   2.172875e+06  2.571004e+06        7.787908e+02        7.037985e+03   
std    2.015015e+07  2.036777e+07        4.787979e+04        9.582593e+04   
min    9.000000e+01  7.337000e+01        0.000000e+00        0.000000e+00   
25%    4.562415e+04  5.401351e+04        0.000000e+00        0.000000e+00   
50%    2.021051e+05  2.416310e+05        0.000000e+00        0.000000e+00   
75%    9.465821e+05  1.128871e+06        0.000000e+00        0.000000e+00   
max    2.160000e+09  1.701000e+09        6.273765e+06        7.380000e+06   

       fl_site_deductible  point_latitude  point_longitude  
count        36634.000000    36634.000000     36634.000000  
mean           192.453210       28.087477       -81.903565  
std           8037.713716        1.647734         1.809066  
min              0.000000       24.547514       -87.447290  
25%              0.000000       26.471996       -82.439331  
50%              0.000000       28.057077       -81.585700  
75%              0.000000       29.623005       -80.372910  
max         450000.000000       30.989820       -80.033257

In [80]:
# to get the description for the categorical columns by all
data.describe(include='all')

policyID statecode             county  eq_site_limit  \
count    36634.000000     36634              36634   3.663400e+04   
unique            NaN         1                 67            NaN   
top               NaN        FL  MIAMI DADE COUNTY            NaN   
freq              NaN     36634               4315            NaN   
mean    548661.937189       NaN                NaN   7.314780e+05   
std     259912.926494       NaN                NaN   1.752300e+07   
min     100074.000000       NaN                NaN   0.000000e+00   
25%     323647.750000       NaN                NaN   0.000000e+00   
50%     548525.000000       NaN                NaN   0.000000e+00   
75%     774252.000000       NaN                NaN   0.000000e+00   
max     999971.000000       NaN                NaN   2.160000e+09   

        hu_site_limit  fl_site_limit  fr_site_limit      tiv_2011  \
count    3.663400e+04   3.663400e+04   3.663400e+04  3.663400e+04   
unique            NaN            NaN            NaN           NaN   
top               NaN            NaN            NaN           NaN   
freq              NaN            NaN            NaN           NaN   
mean     2.074348e+06   6.646009e+05   9.911724e+05  2.172875e+06   
std      1.964150e+07   1.746114e+07   1.767178e+07  2.015015e+07   
min      0.000000e+00   0.000000e+00   0.000000e+00  9.000000e+01   
25%      4.248106e+04   0.000000e+00   0.000000e+00  4.562415e+04   
50%      1.926911e+05   0.000000e+00   0.000000e+00  2.021051e+05   
75%      9.163381e+05   0.000000e+00   5.741010e+04  9.465821e+05   
max      2.160000e+09   2.160000e+09   2.160000e+09  2.160000e+09   

            tiv_2012  eq_site_deductible  hu_site_deductible  \
count   3.663400e+04        3.663400e+04        3.663400e+04   
unique           NaN                 NaN                 NaN   
top              NaN                 NaN                 NaN   
freq             NaN                 NaN                 NaN   
mean    2.571004e+06        7.787908e+02        7.037985e+03   
std     2.036777e+07        4.787979e+04        9.582593e+04   
min     7.337000e+01        0.000000e+00        0.000000e+00   
25%     5.401351e+04        0.000000e+00        0.000000e+00   
50%     2.416310e+05        0.000000e+00        0.000000e+00   
75%     1.128871e+06        0.000000e+00        0.000000e+00   
max     1.701000e+09        6.273765e+06        7.380000e+06   

        fl_site_deductible  fr_site_deductible  point_latitude  \
count         36634.000000        36634.000000    36634.000000   
unique                 NaN                 NaN             NaN   
top                    NaN                 NaN             NaN   
freq                   NaN                 NaN             NaN   
mean            192.453210           26.483594       28.087477   
std            8037.713716         4702.740932        1.647734   
min               0.000000            0.000000       24.547514   
25%               0.000000            0.000000       26.471996   
50%               0.000000            0.000000       28.057077   
75%               0.000000            0.000000       29.623005   
max          450000.000000       900000.000000       30.989820   

        point_longitude         line construction  point_granularity  
count      36634.000000        36634        36634       36634.000000  
unique              NaN            2            5                NaN  
top                 NaN  Residential         Wood                NaN  
freq                NaN        30838        21581                NaN  
mean         -81.903565          NaN          NaN           1.640907  
std            1.809066          NaN          NaN           1.070561  
min          -87.447290          NaN          NaN           1.000000  
25%          -82.439331          NaN          NaN           1.000000  
50%          -81.585700          NaN          NaN           1.000000  
75%          -80.372910          NaN          NaN           3.000000  
max         

slicing , subsetting , Bollean Indexing :
- position based
    - rows --> 0 - 36633
    - columns --> 0 - 17
- label baesd indexing(loc)
    - rows -->0 - 36633
    - columns --> ['policyID', 'statecode', 'county', 'eq_site_limit', 'hu_site_limit',
       'fl_site_limit', 'fr_site_limit', 'tiv_2011', 'tiv_2012',
       'eq_site_deductible', 'hu_site_deductible', 'fl_site_deductible',
       'fr_site_deductible', 'point_latitude', 'point_longitude', 'line',
       'construction', 'point_granularity']

In [84]:
# row with index 1 and all the columns
data.iloc[1,:]

policyID                   448094
statecode                      FL
county                CLAY COUNTY
eq_site_limit         1.32238e+06
hu_site_limit         1.32238e+06
fl_site_limit         1.32238e+06
fr_site_limit         1.32238e+06
tiv_2011              1.32238e+06
tiv_2012              1.43816e+06
eq_site_deductible              0
hu_site_deductible              0
fl_site_deductible              0
fr_site_deductible              0
point_latitude            30.0639
point_longitude          -81.7077
line                  Residential
construction              Masonry
point_granularity               3
Name: 1, dtype: object

In [85]:
data.loc[1,:]

policyID                   448094
statecode                      FL
county                CLAY COUNTY
eq_site_limit         1.32238e+06
hu_site_limit         1.32238e+06
fl_site_limit         1.32238e+06
fr_site_limit         1.32238e+06
tiv_2011              1.32238e+06
tiv_2012              1.43816e+06
eq_site_deductible              0
hu_site_deductible              0
fl_site_deductible              0
fr_site_deductible              0
point_latitude            30.0639
point_longitude          -81.7077
line                  Residential
construction              Masonry
point_granularity               3
Name: 1, dtype: object

In [87]:
# row with index 1 and column with index 1
data.iloc[1,1]

'FL'

In [88]:
data.loc[1,"statecode"]

'FL'

In [91]:
# row with index 1 and columns with index 1,2
# stop value is not necessary
data.iloc[1,1:3]

statecode             FL
county       CLAY COUNTY
Name: 1, dtype: object

In [92]:
# stop value is included
data.loc[1,"statecode":"county"]

statecode             FL
county       CLAY COUNTY
Name: 1, dtype: object

In [94]:
# rows with indices 1,2,3
# columns with indicees 1,2,3,4,5
data.iloc[1:4,1:6]

statecode       county  eq_site_limit  hu_site_limit  fl_site_limit
1        FL  CLAY COUNTY      1322376.3     1322376.30      1322376.3
2        FL  CLAY COUNTY       190724.4      190724.40       190724.4
3        FL  CLAY COUNTY            0.0       79520.76            0.0

In [95]:
data.loc[1:3,"statecode":"county"]

statecode       county
1        FL  CLAY COUNTY
2        FL  CLAY COUNTY
3        FL  CLAY COUNTY